docs/internal/GeneralArchitectureGuide.md

```diff
@@ -6,6 +6,60 @@
+[local volatile variable]: https://github.com/elastic/elasticsearch/blob/v8.13.2/server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java#L123
+[ClusterSettings#initializeAndWatch()]: https://github.com/elastic/elasticsearch/blob/v8.13.2/server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java#L145
+
+[ClusterSettings][] tracks the [core Elasticsearch settings][]. Ultimately the `ClusterSettings` get loaded via the
+[SettingsModule][]. Additional settings from the various plugins are [collected during node construction] and passed into the
+[SettingsModule constructor][]. The Plugin interface has a [getSettings()][] method via which each plugin can declare additional
+settings.
+
+[ClusterSettings]: https://github.com/elastic/elasticsearch/blob/v8.13.2/server/src/main/java/org/elasticsearch/common/settings/ClusterSettings.java#L138
+[core Elasticsearch settings]: https://github.com/elastic/elasticsearch/blob/v8.13.2/server/src/main/java/org/elasticsearch/common/settings/ClusterSettings.java#L204-L586
+[SettingsModule]: https://github.com/elastic/elasticsearch/blob/v8.13.2/server/src/main/java/org/elasticsearch/common/settings/SettingsModule.java#L54
+[collected during node construction]: https://github.com/elastic/elasticsearch/blob/v8.13.2/server/src/main/java/org/elasticsearch/node/NodeConstruction.java#L483
+[SettingsModule constructor]: https://github.com/elastic/elasticsearch/blob/v8.13.2/server/src/main/java/org/elasticsearch/node/NodeConstruction.java#L491-L495
+[getSettings()]: https://github.com/elastic/elasticsearch/blob/v8.13.2/server/src/main/java/org/elasticsearch/plugins/Plugin.java#L203-L208
+
+### Dynamically updating a Setting
+
+`AbstractScopedSettings` has various helper methods to track dynamic changes: it keeps a [registry of `SettingUpdater`][]
+lambdas to run updates when settings are changed in the cluster state. The `ClusterApplierService` also [sends setting updates][]
+through to the `AbstractScopedSettings` that invokes the registered consumers for each setting.
+
+[registry of `SettingUpdater`]: https://github.com/elastic/elasticsearch/blob/v8.13.2/server/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java#L379-L381
+[sends setting updates]: https://github.com/elastic/elasticsearch/blob/v8.13.2/server/src/main/java/org/elasticsearch/cluster/service/ClusterApplierService.java#L490-L494
+
+[TransportClusterUpdateSettingsAction][] and [TransportUpdateSettingsAction][] (and the corresponding REST endpoints) allow users
+to update cluster and index settings, respectively. Settings can be either persisted or transient depending on how they are tied
+to [Metadata][] ([applied here][]) and [IndexMetadata][]. Changes to persisted settings will survive server restart; whereas a
```

Comment:
All _index_ settings are persistent - there is no transient index metadata. The transient/persistent divide only applies to cluster settings.